src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -239,7 +239,7 @@ public virtual void Execute()
 #else
             ApartmentState currentApartment = Thread.CurrentThread.GetApartmentState();
 
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
+            if (Test.RequiresThread || (Test is TestMethod || Test is ParameterizedMethodSuite) && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
                 RunTestOnOwnThread(timeout, TargetApartment);
```

Comment:
This forces the entire suite of test cases to run on a separate thread. It's not what we want. See general comment.